lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_authorizer.rb

```diff
@@ -11,10 +11,24 @@ class GeoEngineer::Resources::AwsApiGatewayAuthorizer < GeoEngineer::Resource
 
   validate -> { validate_required_attributes([:authorizer_uri, :name, :rest_api_id]) }
 
+  after :initialize, -> { self.rest_api_id = _rest_api.to_ref }
```

Comment:
Hmm, what is the difference between the required attribute `rest_api_id` and `_rest_api.to_ref`? It seems like`rest_api_id` should be the source of truth, not entirely sure. Where does `_rest_api` come from again?

Reply:
@itstehkman see all the other api gateway resources. They follow this pattern to pass in the referenced rest api and this comment https://github.com/coinbase/geoengineer/blob/master/lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_method.rb#L20

Reply:
Also, _rest_api is how the rest_api_id is derived. All the gateway resources have standardized passing in the reference to rest_api object via _rest_api

Reply:
Ah ok, thanks for clarifying! As long as the validation on line 12 for rest_api_id is the as the one from _rest_api, then this LGTM.